sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/AvroSerializerProviders.java

```diff
@@ -47,13 +45,43 @@ private static synchronized void load() {
      * Creates an Avro serializer instance based using the first {@link AvroSerializerProvider} found on the classpath.
      *
      * @param schema Schema tied to the Avro serializer for its lifetime.
-     * @return A new Avro serializer instance tied to the passed schema.
+     * @return A new {@link AvroSerializer} instance tied to the passed schema.
      */
-    public static ObjectSerializer createInstance(String schema) {
+    public static AvroSerializer createInstance(String schema) {
         if (defaultProvider == null) {
             load();
         }
 
         return defaultProvider.createInstance(schema);
     }
+
+    /**
+     * Returns the Avro schema for specified object.
+     *
+     * @param object The object having its Avro schema retrieved.
+     * @return The Avro schema for the object.
+     * @throws IllegalArgumentException If the object is an unsupported type.
+     */
+    public static String getSchema(Object object) {
```

Comment:
Why do we need these static methods here - getSchema and getSchemaName? I think callers should just call `createInstance().getSchema()`.

Reply:
Schema handling is a concept that is a layer above the serializer. That is why I added it to the providers instance of the instance level serializer class.